### ISSUE TITLE
[a11y]: Timepicker:  Screen reader should announce the identified error informtion which appears on the screen when user input some invalid data.

### ISSUE THREAD

**Issue by chaagga2** [carbon-design-system/carbon#19596]
### Package

@carbon/react

### Package version

@carbon/react@1.84.0

### Automated testing tool and ruleset

JAWS

### Assistive technology

JAWS

### Description

Issue description: Identified error information is not read by the screen reader. When user navigated on to the edit field and input some invalid information. Screen reader reamins silent.

Recommended fix: Screen reader should announce the identified error informtion which appears on the screen when user input some invalid data. For ex: Screen reader should announce as 'Invalid time format'.
Note: Screen reader should provided identified error information when user renavigated on to the error field.

### Reproduction/example

It can be reproduced on https://cloud.ibm.com/infrastructure/statusHistory
Click the filter icon

<img width="754" alt="Image" src="https://github.com/user-attachments/assets/51551668-0fde-4908-96ec-29120c1ce093" />

### Steps to reproduce

Click filter icon

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by riddhybansal** [comment]
Hey, thanks for reporting this accessibility issue!
I can confirm this occurs in Storybook as well - the time-picker component doesn't announce invalid state messages to screen readers like TextInput and TextArea do. When users input invalid data, JAWS/V0 should announce the error text (e.g., "Invalid time format") but currently remains silent.
